### PR TITLE
Move algorithms to tag_fallback_invoke

### PR DESCRIPTION
**Pull request by toktarev**
## Proposed Changes

  Move algorithms to tag_fallback_invoke
  
## Any background context you want to provide?

https://github.com/STEllAR-GROUP/hpx/issues/5204

### CODE REVIEW

**Review by msimberg — CHANGES_REQUESTED** [review]
Thanks! Could you change the includes as well from `hpx/functional/tag_invoke.hpp` to `hpx/functional/tag_fallback_invoke.hpp`?

**Comment by toktarev** [comment]
> hpx/functional/tag_fallback_invoke.hpp
Sure, Done

**Comment by msimberg** [comment]
retest

**Comment by toktarev** [comment]
> retest

Sorry, what does that mean? Something I should do ?

**Comment by msimberg** [comment]
> > retest
> 
> Sorry, what does that mean? Something I should do ?

That's for Jenkins (CI), nothing to worry about. Perhaps we should change the trigger phrase to something less cryptic. You're not the first one to be confused by it!

**Comment by toktarev** [comment]
We should probably set labels from the previous (corrupted PR)
Not sure if I have such permissions.

**Review by hkaiser — DISMISSED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
@toktarev Thanks for looking into this! One of the main reasons of asking students to look into this issue was for them to understand `tag_invoke` as a concept. Please make sure you understand the point of this change.

**Comment by toktarev** [comment]
@hkaiser , sure, thanks.

I am going to dive deeply inside HPX and contribute here more serious things in the future (if nobody minds of course).

**Comment by toktarev** [comment]
https://www.youtube.com/watch?v=Hi5WnRNckm4 - amazing video.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/is_sorted.hpp [hunk 1]
```diff
@@ -447,7 +447,7 @@ namespace hpx { namespace parallel { inline namespace v1 {
```
Comment (msimberg) [inline libs/parallelism/algorithms/include/hpx/parallel/algorithms/is_sorted.hpp:447]: Would you mind adding the `tag_fallback_invoke` include to this file. It was already missing the `tag_invoke` include.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/mismatch.hpp [hunk 2]
```diff
@@ -496,7 +496,7 @@ namespace hpx {
     ///////////////////////////////////////////////////////////////////////////
```
Comment (msimberg) [inline libs/parallelism/algorithms/include/hpx/parallel/algorithms/mismatch.hpp:496]: Same thing here.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/move.hpp [hunk 3]
```diff
@@ -191,7 +191,7 @@ namespace hpx {
     ///////////////////////////////////////////////////////////////////////////
```
Comment (msimberg) [inline libs/parallelism/algorithms/include/hpx/parallel/algorithms/move.hpp:191]: Same here.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove.hpp [hunk 4]
```diff
@@ -466,7 +466,7 @@ namespace hpx {
     ///////////////////////////////////////////////////////////////////////////
```
Comment (msimberg) [inline libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove.hpp:466]: Same here.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove_copy.hpp [hunk 5]
```diff
@@ -454,7 +454,7 @@ namespace hpx {
     ///////////////////////////////////////////////////////////////////////////
```
Comment (msimberg) [inline libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove_copy.hpp:454]: Same here.

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/search.hpp [hunk 6]
```diff
@@ -368,7 +368,7 @@ namespace hpx { namespace parallel { inline namespace v1 {
 namespace hpx {
```
Comment (msimberg) [inline libs/parallelism/algorithms/include/hpx/parallel/algorithms/search.hpp:368]: And here.

**Review by msimberg — CHANGES_REQUESTED** [review]
Looks good!

There were a few files that didn't have the `tag_invoke` include in the first place. Would you mind adding the `tag_fallback_invoke` include to those files while you're at it?

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/is_sorted.hpp (hunk 1 above)
Comment (toktarev) [reply libs/parallelism/algorithms/include/hpx/parallel/algorithms/is_sorted.hpp:448]: done

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/mismatch.hpp (hunk 2 above)
Comment (toktarev) [reply libs/parallelism/algorithms/include/hpx/parallel/algorithms/mismatch.hpp:497]: done

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/move.hpp (hunk 3 above)
Comment (toktarev) [reply libs/parallelism/algorithms/include/hpx/parallel/algorithms/move.hpp:191]: done

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove.hpp (hunk 4 above)
Comment (toktarev) [reply libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove.hpp:467]: done

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove_copy.hpp (hunk 5 above)
Comment (toktarev) [reply libs/parallelism/algorithms/include/hpx/parallel/algorithms/remove_copy.hpp:455]: done

File: libs/parallelism/algorithms/include/hpx/parallel/algorithms/search.hpp (hunk 6 above)
Comment (toktarev) [reply libs/parallelism/algorithms/include/hpx/parallel/algorithms/search.hpp:368]: done

**Comment by msimberg** [comment]
retest

**Pull request merged**